src/SimpleSAML/Metadata/MetaDataStorageHandlerPdo.php

```diff
@@ -159,16 +159,10 @@ public function getMetaData(string $entityId, string $set): ?array
         }
 
         $tableName = $this->getTableName($set);
-
-        // according to the docs, it looks like *-idp-hosted metadata are the types
-        // that allow the __DYNAMIC:*__ entity id.  with the current table design
-        // we need to lookup the specific metadata entry but also we need to lookup
-        // any dynamic entries to see if the dynamic hosted entity id matches
         if (substr($set, -10) == 'idp-hosted') {
```

Comment:
This is no longer needed since there's no distinction in the lookup between idp-hosted and other types. Can use just one query. And that's the one with exact match, not LIKE

Reply:
Removed it